proposed/http-message.md

```diff
@@ -1260,6 +1280,15 @@ interface UriInterface
      *
      * The string returned MUST omit the leading "#" character.
      *
+     * The value returned MUST be percent-encoded, but MUST NOT double-encode
+     * any characters. To determine what characters to encode, please refer to
+     * RFC 3986, Sections 2 and 3.5.
+     *
+     * As an example, if the fragment should include a hash symbol ("#"), that
+     * value MUST be encoded (e.g., "%23") when passed to the instance.
```

Comment:
This seems wrong to me. For the fragment, a  hash is always wrong inside the fragment. So it must be encoded anyway. So it does not need to be passed encoded to `withFragment` since it should be encoded automatically anyway. So `withFragment('foo#bar')` should be equal to `withFragment('foo%23bar')`.

The examples for path and query are correct because both version exist: encoded and non-encoded. This is why the caller must decide what he means.